### PR TITLE
Knife webbing fixes and consistency

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This PR:

Adds sounds for inserting and removing knifes to the knife webbing

Adds draw delay for knives to the knife webbing

Allows using quickdraw with the knife webbing

Standardizes knife draw delay in a define

Knife webbing is now spawned full

Allows knife webbing to be alt-clicked to draw from it

Reviewer notes:
The gun helper stuff looks weird to me. I don't think we ever have to iterate through internal directly from a mob's accessories var but there may be some snowflake stuff going on. Happy to move things around but leaving just in case. (Also looks out of scope :D)

# Explain why it's good for the game

Consistency and usability good. I genuinely have no clue how I ended up in this rabbit hole.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow
add: Added sounds for inserting and removing knifes to the knife webbing
add: Added draw delay for knives to the knife webbing
add: Allowed using quickdraw with the knife webbing
add: Allowed knife webbing to be alt-clicked to draw from it
add: Knife webbing is now spawned full
code: Standardized knife draw delay in a define
/:cl:

### CODE REVIEW

File: code/modules/clothing/under/ties.dm [hunk 1]
```diff
@@ -603,6 +609,31 @@
 		/obj/item/attachable/bayonet,
 		/obj/item/weapon/throwing_knife,
 	)
+	storage_flags = STORAGE_ALLOW_QUICKDRAW|STORAGE_FLAGS_POUCH
+	var/draw_cooldown = 0
```
Comment (harryob) [inline code/modules/clothing/under/ties.dm:613]: cool-down defines

File: code/modules/clothing/under/ties.dm [hunk 2]
```diff
@@ -603,6 +609,31 @@
...
+
+/obj/item/storage/internal/accessory/knifeharness/fill_preset_inventory()
+	for(var/i = 1 to storage_slots)
+		new /obj/item/weapon/throwing_knife(src)
+
+/obj/item/storage/internal/accessory/knifeharness/attack_hand(mob/user, mods)
+	. = ..()
+
+	if(draw_cooldown >= world.time)
```
Comment (harryob) [inline code/modules/clothing/under/ties.dm:622]: cooldown def

File: code/modules/clothing/under/ties.dm [hunk 3]
```diff
@@ -603,6 +609,31 @@
...
+		to_chat(user, SPAN_WARNING("You need to wait before drawing another knife!"))
+		return FALSE
+
+	if(length(contents))
+		contents[length(contents)].attack_hand(user, mods)
+		draw_cooldown = world.time + BAYONET_DRAW_DELAY
```
Comment (harryob) [inline code/modules/clothing/under/ties.dm:628]: cooldown defines

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/clothing/under/ties.dm (hunk 1 above)
Comment (morrowwolf) [reply code/modules/clothing/under/ties.dm:613]: *screams

File: code/modules/clothing/under/ties.dm [hunk 4]
```diff
@@ -603,6 +609,31 @@
...
+/obj/item/storage/internal/accessory/knifeharness/_item_insertion(obj/item/inserted_item, prevent_warning = 0)
+	..()
+	playsound(src, 'sound/weapons/gun_shotgun_shell_insert.ogg', 15, TRUE)
+
+/obj/item/storage/internal/accessory/knifeharness/_item_removal(obj/item/removed_item, atom/new_location)
+	..()
+	playsound(src, 'sound/weapons/gun_shotgun_shell_insert.ogg', 15, TRUE)
 
 /obj/item/clothing/accessory/storage/knifeharness/duelling
 	name = "decorated harness"
```
Comment (realforest2001) [inline code/modules/clothing/under/ties.dm:639]: While you're doing this, the decorated harness can't actually hold the knives it spawns with. They're larger than the main ones, so once removed from harness they cannot be replaced.

**Pull request merged**